lib/UnsafeCachePlugin.js

```diff
@@ -1,41 +1,41 @@
+"use strict";
 /*
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var createInnerCallback = require("./createInnerCallback");
+const createInnerCallback = require("./createInnerCallback");
 
-function UnsafeCachePlugin(source, filterPredicate, cache, target) {
-	this.source = source;
-	this.filterPredicate = filterPredicate;
-	this.cache = cache || {};
-	this.target = target;
-}
-module.exports = UnsafeCachePlugin;
+class UnsafeCachePlugin {
+	constructor(source, filterPredicate, cache, target) {
+		this.source = source;
+		this.filterPredicate = filterPredicate;
+		this.cache = cache || {};
+		this.target = target;
+	}
 
-function getCacheId(request) {
-	return JSON.stringify({
-		context: request.context,
-		path: request.path,
-		query: request.query,
-		request: request.request
-	});
+	apply(resolver) {
+		const filterPredicate = this.filterPredicate;
+		const cache = this.cache;
+		const target = this.target;
+		resolver.plugin(this.source, (request, callback) => {
+			if(!filterPredicate(request)) return callback();
+			const cacheId = getCacheId(request);
+			const cacheEntry = cache[cacheId];
+			if(cacheEntry) return callback(null, cacheEntry);
+			resolver.doResolve(target, request, null, createInnerCallback((err, result) => {
+				if(err) return callback(err);
+				if(result) return callback(null, cache[cacheId] = result);
+				callback();
+			}, callback));
+		});
+	}
 }
 
-UnsafeCachePlugin.prototype.apply = function(resolver) {
-	var filterPredicate = this.filterPredicate;
-	var cache = this.cache;
-	var target = this.target;
-	resolver.plugin(this.source, function(request, callback) {
-		if(!filterPredicate(request)) return callback();
-		var cacheId = getCacheId(request);
-		var cacheEntry = cache[cacheId];
-		if(cacheEntry) {
-			return callback(null, cacheEntry);
-		}
-		resolver.doResolve(target, request, null, createInnerCallback(function(err, result) {
-			if(err) return callback(err);
-			if(result) return callback(null, cache[cacheId] = result);
-			callback();
-		}, callback));
-	});
-};
+module.exports = UnsafeCachePlugin;
+
+const getCacheId = request => JSON.stringify({
```

Comment:
@sokra is this a named function you rely on for debugging? Otherwise I'm fine with this going anon (i guess chrome binds to the identifier's name anyways for stack debugging as of later versions of Chrome.